### PR TITLE
Define NATIVE_WIN32_PYTHON environment variable

### PR DESCRIPTION
**Pull request by UK992**
r? @Wafflespeanut or @larsbergstrom 

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13650)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Firstly, I'm curious why we need this variable. And second, why can't we use `distutils.spawn.find_executable`?

**Comment by UK992** [comment]
This variable is needed for mozjs_sys in case if native python is not installed on default location (`C:/Python27`).
`distutils.spawn.find_executable` it's useless in this case, because it will pick mingw python.

File: python/servo/command_base.py [hunk 1]
```diff
@@ -205,6 +205,21 @@ def is_macosx():
     return sys.platform == 'darwin'
 
 
+def find_native_win32_python():
+    paths = subprocess.Popen(['where', 'python.exe'],
```
Comment (Wafflespeanut) [inline python/servo/command_base.py]: I'm pretty sure `which` is present in most of the distributions. I'm curious whether we can rely on `where`

File: python/servo/command_base.py [hunk 2]
```diff
@@ -205,6 +205,21 @@ def is_macosx():
...
+                             shell=False, stdout=subprocess.PIPE).stdout.read()
+
+    for p in paths.splitlines():
+        p = p.replace("\\", "/")
+        py_type = subprocess.Popen([p, '-c', 'import sys; print(sys.platform)'],
+                                   shell=False, stdout=subprocess.PIPE).stdout.read()
+        if "win32" in py_type and "/mingw64/bin" not in p:
```
Comment (Wafflespeanut) [inline python/servo/command_base.py]: We could just check for the presence of `"mingw64"`, instead of going for a path?

**Comment by Wafflespeanut** [comment]
Just a couple of comments. Otherwise LGTM. Although, r? @larsbergstrom

File: python/servo/command_base.py (hunk 1 above)
Comment (Wafflespeanut) [inline python/servo/command_base.py]: Oh, TIL :)

File: python/servo/command_base.py (hunk 2 above)
Comment (UK992) [inline python/servo/command_base.py]: done

**Comment by larsbergstrom** [comment]
@bors-servo retry
- infra

**Comment by larsbergstrom** [comment]
```

TypeError: execve() arg 3 contains a non-string value

  File "/c/buildbot/slave/windows-dev/build/python/servo/build_commands.py", line 256, in build
    env=env, cwd=self.servo_crate(), verbose=verbose)
  File "/c/buildbot/slave/windows-dev/build/python/servo/command_base.py", line 167, in call
    return subprocess.call(*args, shell=sys.platform == 'win32', **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
program finished with exit code 1
elapsedTime=50.527000
```

**Comment by UK992** [comment]
Updated.

**Pull request merged**